OpenMarket/OpenMarket/Application/Presentation/ViewController/ProductList/ProductListViewController.swift

```diff
@@ -201,90 +209,161 @@ final class MarketProductsViewController: UIViewController {
+        resetData()
+    }
+    
+    @objc private func gridHandleRefreshControl() {
+        resetData()
+    }
+    
     @objc private func didSegmentedControlTapped(_ segment: UISegmentedControl) {
         self.shouldHideListView = segment.selectedSegmentIndex != 0
     }
     
     @objc private func addButtonTapped(_ sender: UIBarButtonItem) {
-        let productEnrollmentViewController = UINavigationController(rootViewController: ProductEnrollmentViewController())
-        productEnrollmentViewController.modalPresentationStyle = .fullScreen
+        let productEnrollmentViewController = ProductEnrollmentViewController()
         
-        self.present(productEnrollmentViewController, animated: true)
+        DispatchQueue.main.async { [weak self] in
+            self?.present(viewController: productEnrollmentViewController)
+        }
     }
 }
 
 // MARK: - MarketProductsViewDelegate
 
-extension MarketProductsViewController: MarketProductsViewControllerDelegate {
-    func didReceiveResponse(_ view: MarketProductsViewController.Type,
-                            by data: ProductListEntity) {
-        updateUI(by: data)
+extension ProductListViewController: ProductListDelegate {
+    func productListViewController(_ view: ProductListViewController.Type,
+                                   didRecieve productListInfo: ProductListEntity) {
+        DispatchQueue.main.async { [weak self] in
+            self?.setUpCollectionViewFor(hiding: false)
+            LoadingIndicator.hideLoading()
+            self?.updateUI(by: productListInfo)
+        }
     }
 }
 
 // MARK: - UICollectionViewDelegate
 
-extension MarketProductsViewController: UICollectionViewDelegate {
+extension ProductListViewController: UICollectionViewDelegate {
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        guard let productID = shouldHideListView ? self.listDataSource?.itemIdentifier(for: indexPath)?.id : self.gridDataSource?.itemIdentifier(for: indexPath)?.id else {
-            collectionView.deselectItem(at: indexPath, animated: true)
+        guard let product = shouldHideListView
+                ? listDataSource?.itemIdentifier(for: indexPath)
+                : gridDataSource?.itemIdentifier(for: indexPath) else {
+            collectionView.deselectItem(at: indexPath,
+                                        animated: true)
             return
         }
         
-        let productDetailViewController = ProductDetailViewController()
-        productDetailViewController.productID = productID
-        self.navigationController?.pushViewController(productDetailViewController, animated: true)
+        let productDetailViewController = ProductDetailsViewController()
+        productDetailViewController.productID = product.id
+        productDetailViewController.productVendorID = product.vendorID
+        productDetailViewController.title = product.name
```

Comment:
```suggestion
        let productDetailViewController: ProductDetailsViewController = {
            let viewController = ProductDetailsViewController()
            viewController.productID = product.id
            viewController.productVendorID = product.vendorID
            viewController.title = product.name
            return viewController
        }()
```
=> 이런식으로 바꾸면 좀 더 응집되어 보여 좋다고 생각합니다. 참고차만 말씀드린것이지 안 바꾸셔도 좋습니다. 
=> 추후에 전수열님의 [Then](https://github.com/devxoul/Then) 라이브러리를 한번 써보는 것도 개인적으로 추천드려요 :)